Code/GraphMol/SubstructLibrary/Wrap/rough_test.py

```diff
@@ -73,86 +73,109 @@ def setUp(self):
     pass
 
   def test0SubstructLibrary(self):
-    for fpholderCls in [None, rdSubstructLibrary.PatternHolder]:
-      for holder in [rdSubstructLibrary.MolHolder(), rdSubstructLibrary.CachedMolHolder(),
-                     rdSubstructLibrary.CachedSmilesMolHolder()]:
-        if fpholderCls: fpholder = fpholderCls()
-        else: fpholder = None
-        slib_ = rdSubstructLibrary.SubstructLibrary(holder, fpholder)
-        for i in range(100):
-          m = Chem.MolFromSmiles("c1ccccc1")
-          self.assertEqual(slib_.AddMol(m), i)
-
-        libs = [slib_]
-        if rdSubstructLibrary.SubstructLibraryCanSerialize():
-          serialized1 = pickle.loads(pickle.dumps(slib_))
-          serialized2 = rdSubstructLibrary.SubstructLibrary(slib_.Serialize())
-          libs.append(serialized1)
-          libs.append(serialized2)
-
-        for slib in libs:
-          res = slib.GetMatches(m)
-          t2 = time.time()
-          self.assertTrue(len(res) == 100)
-
-          res = slib.GetMatches(m)
-
-          self.assertEqual(len(res), 100)
-          self.assertTrue(set(res) == set(list(range(100))))
-
-          res = slib.GetMatches(m, maxResults=100);
-          self.assertEqual(len(res), 100)
-          self.assertEqual(len(slib.GetMatches(m, startIdx=0, endIdx=100)), 100)
-
-          self.assertTrue(slib.HasMatch(m))
-          self.assertEqual(slib.CountMatches(m), 100)
+    for keyholderCls in [None, rdSubstructLibrary.KeyFromPropHolder]:
```

Comment:
Although these tests look completely different to git, we only added the outer loop for the key holder and added some checks for the keys themselves.